conda_build/metadata.py

```diff
@@ -994,7 +994,7 @@ def get_output_dicts_from_metadata(
     metadata: MetaData,
     outputs: list[dict[str, Any]] | None = None,
 ) -> list[dict[str, Any]]:
-    outputs = outputs or metadata.get_section("outputs")
```

Comment:
Why do we need the identity comprehension? `get_section` checks for `outputs` in `OPTIONALLY_ITERABLE_FIELDS`, and always returns a list for those 🤔, even if empty.

Reply:
See https://github.com/conda/conda-build/blob/1bbc9d93288df7e019a4dc949f3e71835fed8acf/conda_build/metadata.py#L1481-L1495

Reply:
We are making a copy of the list. Otherwise this function modifies the list in place and causes a bug.

Reply:
I removed the "or []" if that is what you meant.

Reply:
Ah, ok, I get it now. I would have used `.copy()` here to make the intent clearer